CraftsmenLtd/BloodConnect
# I-#231: Implemented User Profile Context & Conditional Profile Update Routing After SSO+Manual Login

Fixes #231

## What kind of change does this PR introduce (check at least one)

## How would you describe the changes in a changelog (1-2 lines)?
Implemented conditional profile update routing after SSO login using React Context. Users are now only redirected to update their profile on first-time login, with subsequent logins going directly to the main app. Aslo fixed and refactored some other issues mentioned in ticket description.

## How to test

Test scenarios:
1. First-time SSO login:
   - Login via Google/Facebook SSO
   - Verify redirect to ADD_PERSONAL_INFO route
   - Complete profile information
   - Verify redirect to BOTTOM_TABS

2. Subsequent SSO logins:
   - Logout and login again via SSO
   - Verify direct redirect to BOTTOM_TABS
   - Verify profile information persists

3. Same flow followed in manual login also.

4. Context implementation:
   - Verify UserProfileContext provides profile data
   - Check profile data accessibility across components

## Checklist

## Reviewer Request
- N/A

## Additional Notes
Implementation highlights:
- Created UserProfileContext for centralized profile management
- Added bloodGroup existence check for routing logic
- Implemented profile data caching to reduce API calls
- Added error handling for profile fetch failures
- Maintained clean architecture separation of concerns

## Deploy risk & follow up
Justification: Changes are isolated to frontend routing logic and context implementation. No backend API changes required. Easy rollback if needed.
